r/tests/testthat/test-dataset.R

```diff
@@ -303,11 +303,54 @@ test_that("Other text delimited dataset", {
       filter(integer > 6) %>%
       summarize(mean = mean(integer))
   )
+})
+
+test_that("readr parse options", {
+  arrow_opts <- names(formals(CsvParseOptions$create))
+  readr_opts <- names(formals(readr_to_csv_parse_options))
+
+  # Arrow and readr options are mutually exclusive
+  expect_equal(
```

Comment:
Why does this need to be asserted? Because the validating code assumes it?

Reply:
The code in `csv_file_format_parse_options()` in `dataset-format.R` will error or behave incorrectly if any of the Arrow file parsing options (`delimiter`, `quoting`, etc.) happens to have exactly the same name as any of the supported readr-style options (`delim`, `quote`, etc.). Some future change might accidentally cause these two lists of option names to overlap, so I added this test which will fail if that happens.

Reply:
Ok, that's what I figured. Could you please add a comment here explaining that?